driver-core/src/main/com/mongodb/internal/operation/CreateCollectionOperation.java

```diff
@@ -407,7 +426,34 @@ private BsonDocument getCommand(final ConnectionDescription description) {
         if (collation != null) {
             document.put("collation", collation.asDocument());
         }
+        if (expireAfter != null) {
+            document.put("expireAfterSeconds", new BsonInt64(expireAfter.getSeconds()));
+        }
+        if (timeSeriesOptions != null) {
+            BsonDocument timeSeriesDocument = new BsonDocument("timeField", new BsonString(timeSeriesOptions.getTimeField()));
+            String metaField = timeSeriesOptions.getMetaField();
+            if (metaField != null) {
+                timeSeriesDocument.put("metaField", new BsonString(metaField));
+            }
+            TimeSeriesGranularity granularity = timeSeriesOptions.getGranularity();
+            if (granularity != null) {
+                timeSeriesDocument.put("granularity", new BsonString(getGranularityAsString(granularity)));
+            }
+            document.put("timeseries", timeSeriesDocument);
+        }
         return document;
     }
 
+    private String getGranularityAsString(final TimeSeriesGranularity granularity) {
```

Comment:
We could push this code into the enum, but generally (though there are exceptions) we have preferred to keep the mapping from public API to command protocol out of the public API)